src/vs/platform/externalTerminal/node/externalTerminalService.ts

```diff
@@ -31,20 +29,12 @@ export class WindowsExternalTerminalService extends ExternalTerminalService impl
 	private static readonly CMD = 'cmd.exe';
 	private static _DEFAULT_TERMINAL_WINDOWS: string;
 
-	constructor(
-		@optional(IConfigurationService) private readonly _configurationService: IConfigurationService
-	) {
-		super();
-	}
-
-	public openTerminal(cwd?: string): Promise<void> {
-		const configuration = this._configurationService.getValue<IExternalTerminalConfiguration>();
+	public openTerminal(configuration: IExternalTerminalSettings, cwd?: string): Promise<void> {
 		return this.spawnTerminal(cp, configuration, processes.getWindowsShell(), cwd);
 	}
 
-	public spawnTerminal(spawner: typeof cp, configuration: IExternalTerminalConfiguration, command: string, cwd?: string): Promise<void> {
-		const terminalConfig = configuration.terminal.external;
```

Comment:
so the other way to fix this would've been `configuration?.terminal?.external`, correct?

Reply:
That would have fixed it but other problems could potentially still happen. The reason it's undefined is because the schema doesn't get registered in the main process, plus workspace settings will never work in the main process if we wanted to add that. So the safest option is to pull it in from the renderer that definitely gets the right settings values.